appinventor/components/src/com/google/appinventor/components/runtime/YandexTranslate.java

```diff
@@ -105,7 +114,7 @@ public YandexTranslate(ComponentContainer container) {
   public void RequestTranslation(final String languageToTranslateTo,
                                  final String textToTranslate) {
 
-    if (yandexKey.equals("")){
+    if ((userYandexKey.equals("") || userYandexKey.equals("DEFAULT")) && yandexKey.equals("")){
```

Comment:
Since `userYandexKey` can be set from outside this component, swap the equals() call so that it is invoked on the string constant on the off chance `userYandexKey == null`.

Reply:
Change made and converted to TextUtils